enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -104,198 +156,160 @@ This is often required for industries such as the US public sector, healthcare o
 * A pipeline shall support at least one output.
 * A pipeline may support multiple outputs.
 * Outputs that require secrets which are not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
-* Secrets shall have keys of: `tls.crt`, `tls.key`, `ca-bundler.crt` which point to the respective certificates for which they represent. Secrets shall have the key `shared_key` for use when using `forward` in a secure manner
-* The operator will only support a single instance of log forwarding named `instance` for at least the tech preview release
-* Log forwarding is enabled for tech preview when the instance of `clusterloggging` is annotated with: `clusterlogging.openshift.io/logforwardingtechpreview: enabled`
+* Forwarder CRs can refer to secrets for TLS certificates (see security section below)
+* The operator allows multiple instances of `forwarder`.
+  * Instances in the `openshift-logging` namespace are "global" and can forward logs from any container.
+  * Instances in other namespaces are namespace-restricted and can only forward logs from their own namespace.
+* Log forwarding is enabled when a `forwarder` object is created, and disabled when no forwarder objects exist.
+* When log forwarding is disabled, default forwarding to the internal elasticsearch instance is enabled.
+* When log forwarding is enabled, default forwarding is disabled. To enable both, create a global forwarder in the openshift-logging namespace with a single pipeline: `pipeline: { output: openshift-logging-default }`
 
 #### Security
-* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: `tls.crt`, `tls.key`, `ca-bundler.crt`
-* The secret must exist in the `openshift-logging` namespace
-* Endpoints defined without a secret will use unsecure communication and must explicitly declare `insecure: true` in the output specifification
+* An output MAY support basic server-authenticated TLS by using a URL scheme that requires TLS, e.g. elasticsearch+https:// syslog+tls:// etc.
+* An output MAY support client-authenticated TLS by declaring a `secret` with keys `tls.crt`, `tls.key`, `ca.crt`
+* An output may support username/password or shared-key authentication either in the `url` or in the `secret`.
+* Outputs that intentionally do not use TLS must have an explicit `insecure: true` to avoid security breaches due to mis-configuration.
 
 #### Log Message Routing
-* Undeclared source types (e.g. `logs.infra`) will be dropped if they are not defined in `spec.pipelines` entry
-* Declared source types are routed to the specified output
-
-For example, If you have specified a pipeline for `logs.app` sources, but did not specify one for `logs.infra`, then `logs.infra` will be dropped
-
+* Pipelines can select multiple container log streams (fan in) by:
+  * Source type (`Infrastructure`, `Application`, `Audit`) 
```

Comment:
is source-type only valid for global forwarders as well?  (i'd expect namespaced forwarders to be restricted to application type only)

Reply:
Correct, and in the new proposal the NamespaceLogForwarder is a separate CRD, like the ClusterLogForwarder but with a few things deleted.